controller/generic/metacontroller.go

```diff
@@ -106,8 +104,28 @@ type ConfigMetaController struct {
 // This follows **functional options** pattern
 type ConfigMetaControllerOption func(*ConfigMetaController) error
 
-// SetMetaControllerConfigLoadFn sets the config loader function
-func SetMetaControllerConfigLoadFn(
+// SetMetacConfigToRetryIndefinitelyForStart will let this
+// controller to retry indefinitely till all its watch controllers
+// are started
+//
+// NOTE:
+//	Indefinite retry is set only when the provided flag is true
+func SetMetacConfigToRetryIndefinitelyForStart(enabled *bool) ConfigMetaControllerOption {
+	return func(c *ConfigMetaController) error {
+		if enabled == nil || !*enabled {
+			return nil
+		}
+		// We are not bothered about performance. We do not
+		// need to be fast. We can be lazy but not to the
+		// extent of sleeping for hours.
+		c.WaitIntervalBetweenStarts = 1 * time.Minute
+		c.RetryIndefinitelyForStart = k8s.BoolPtr(true)
```

Comment:
if I understand correctly, it will set by default to `true`, is this desirable ?

Reply:
This is not the default behaviour.
I have tried to explain in code comments.